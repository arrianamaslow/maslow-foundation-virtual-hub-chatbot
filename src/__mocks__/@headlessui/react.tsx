/**
 * @jest-environment jsdom
 */

import { DialogProps } from '@headlessui/react'
import React, { ReactNode } from 'react'

type DialogPanelTestProps = {
  children?: ReactNode
}

interface DialogTestProps extends DialogProps {
  children?: ReactNode
}

const DialogPanel: React.FC<DialogPanelTestProps> = ({ children }) => {
  return <div data-testid='mock-dialog-panel-component'>{children}</div>
}

const Dialog: React.FC<DialogTestProps> = ({ open, children }) => {
  return (
    <div data-testid='mock-dialog-component'>
      {open ? 'DIALOGUEOPENED' : 'DIALOGUECLOSED'}
      {children}
    </div>
  )
}

export { Dialog, DialogPanel }
