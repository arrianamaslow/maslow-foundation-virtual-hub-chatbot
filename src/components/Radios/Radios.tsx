import React, { useEffect, useState } from 'react'
import CheckBox from '@/components/CheckBox/CheckBox'

export type RadiosProps = {
  radioGroupName: string
  radioValues: string[]
  defaultSelectedIndex?: number
  radioClickedCallback: React.Dispatch<React.SetStateAction<string>>
}

const Radios: React.FC<RadiosProps> = ({
  radioGroupName,
  radioValues,
  defaultSelectedIndex,
  radioClickedCallback
}) => {
  const [selectedIndex, setSelectedIndex] = useState<number | undefined>(defaultSelectedIndex)

  const onRadioChange = (newSelectedIndex: number) => {
    setSelectedIndex(newSelectedIndex)
    radioClickedCallback(radioValues[newSelectedIndex])
  }

  useEffect(() => {
    if (
      !(defaultSelectedIndex === undefined) &&
      defaultSelectedIndex >= 0 &&
      defaultSelectedIndex < radioValues.length
    ) {
      onRadioChange(defaultSelectedIndex)
    }
  }, [])

  return (
    <div className='flex md:h-[20px] min-w-[8rem]'>
      {radioValues.map((radioValue, index) => (
        <div key={index} className=' min-w-[8rem]'>
          <label
            tabIndex={index}
            className='flex gap-2 items-center md:h-[20px]'
            data-testid={`mock-radio-option-${index}`}
          >
            <CheckBox tickIsActive={selectedIndex === index} />
            <input
              type='radio'
              name={radioGroupName}
              className='hidden'
              onChange={(e) => {
                onRadioChange(index)
              }}
              checked={selectedIndex === index}
            />
            {radioValue}
          </label>
        </div>
      ))}
    </div>
  )
}

export default Radios
