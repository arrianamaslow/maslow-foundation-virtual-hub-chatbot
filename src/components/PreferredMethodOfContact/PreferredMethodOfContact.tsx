'use client'
import Radios from '@/components/Radios/Radios'

import React, { useState } from 'react'

const PreferredMethodOfContact: React.FC = () => {
  const [preferredContactMethod, setPreferredContactMethod] = useState<string>('not selected')
  return (
    <div className='text-[1.313rem] w-[23.438rem]'>
      <p className='font-bold'>Preferred method of contact</p>
      <Radios
        radioGroupName='preferredContactMethodRadioButton'
        radioClickedCallback={setPreferredContactMethod}
        radioValues={['Phone', 'Email']}
      />
    </div>
  )
}

export default PreferredMethodOfContact
