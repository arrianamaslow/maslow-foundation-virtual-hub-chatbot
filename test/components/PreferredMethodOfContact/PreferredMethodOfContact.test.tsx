/**
 * @jest-environment jsdom
 */
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import PreferredMethodOfContact from '@/components/PreferredMethodOfContact/PreferredMethodOfContact'
import { RadiosProps } from '@/components/Radios/Radios'

jest.mock('@/components/Radios/Radios', () => {
  const MockRadio = ({ radioGroupName, radioValues, defaultSelectedIndex }: RadiosProps) => (
    <div data-testid='mock-radio'>
      <p data-testid='mock-radio-group-name'>{radioGroupName}</p>
      {radioValues.map((value, index) => (
        <p key={index} data-testid={`mock-radio-option-${index}`}>
          {value}
        </p>
      ))}
      <p data-testid='mock-default-index-selected'>{defaultSelectedIndex}</p>
    </div>
  )

  return MockRadio
})

describe('<PreferredMethodOfContact />', () => {
  it('renders correctly', () => {
    const tree = render(<PreferredMethodOfContact />)
    expect(tree).toMatchSnapshot()
  })

  it('renders radio component', () => {
    render(<PreferredMethodOfContact />)
    expect(screen.getByTestId('mock-radio')).toBeDefined()
  })

  it('passes expected values to <Radio /> component', () => {
    render(<PreferredMethodOfContact />)
    expect(screen.getByTestId('mock-radio-group-name')).toHaveTextContent(
      'preferredContactMethodRadioButton'
    )
    expect(screen.getByTestId('mock-radio-option-0')).toHaveTextContent('Phone')
    expect(screen.getByTestId('mock-radio-option-1')).toHaveTextContent('Email')
    expect(screen.getByTestId('mock-default-index-selected')).toHaveTextContent('')
  })
})
