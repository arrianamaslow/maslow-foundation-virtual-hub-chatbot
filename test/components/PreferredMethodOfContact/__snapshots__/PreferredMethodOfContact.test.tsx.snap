// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PreferredMethodOfContact /> renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="text-[1.313rem] w-[23.438rem]"
      >
        <p
          class="font-bold"
        >
          Preferred method of contact
        </p>
        <div
          data-testid="mock-radio"
        >
          <p
            data-testid="mock-radio-group-name"
          >
            preferredContactMethodRadioButton
          </p>
          <p
            data-testid="mock-radio-option-0"
          >
            Phone
          </p>
          <p
            data-testid="mock-radio-option-1"
          >
            Email
          </p>
          <p
            data-testid="mock-default-index-selected"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="text-[1.313rem] w-[23.438rem]"
    >
      <p
        class="font-bold"
      >
        Preferred method of contact
      </p>
      <div
        data-testid="mock-radio"
      >
        <p
          data-testid="mock-radio-group-name"
        >
          preferredContactMethodRadioButton
        </p>
        <p
          data-testid="mock-radio-option-0"
        >
          Phone
        </p>
        <p
          data-testid="mock-radio-option-1"
        >
          Email
        </p>
        <p
          data-testid="mock-default-index-selected"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
